ci-operator/step-registry/aws/provision/dns-firewall/aws-provision-dns-firewall-commands.sh

```diff
@@ -0,0 +1,186 @@
+  "storage.googleapis.com",
+  "quayio-production-s3.s3.amazonaws.com",
+  "api.openshift.com",
+  "rhcos.mirror.openshift.com"
+')
+
+OSD_DOMAIN_ALLOW_LIST=$(echo -e '
+  "registry.redhat.io",
+  "quay.io",
+  "sso.redhat.com",
+  "quay-registry.s3.amazonaws.com",
+  "ocm-quay-production-s3.s3.amazonaws.com",
+  "quayio-production-s3.s3.amazonaws.com",
+  "cart-rhcos-ci.s3.amazonaws.com",
+  "openshift.org",
+  "registry.access.redhat.com",
+  "registry.connect.redhat.com",
+  "pull.q1w2.quay.rhcloud.com",
+  "q1w2.quay.rhcloud.com",
+  "www.okd.io",
+  "www.redhat.com",
+  "aws.amazon.com",
+  "catalog.redhat.com",
+
+  "cert-api.access.redhat.com",
+  "api.access.redhat.com",
+  "infogw.api.openshift.com",
+  "console.redhat.com",
+  "cloud.redhat.com",
+  "observatorium-mst.api.openshift.com",
+  "observatorium.api.openshift.com",
+
+  "ec2.amazonaws.com",
+  "events.'${REGION}'.amazonaws.com",
+  "iam.amazonaws.com",
+  "route53.amazonaws.com",
+  "sts.amazonaws.com",
+  "sts.'${REGION}'.amazonaws.com",
+  "tagging.us-east-1.amazonaws.com",
+  "ec2.'${REGION}'.amazonaws.com",
+  "elasticloadbalancing.'${REGION}'.amazonaws.com",
+  "servicequotas.'${REGION}'.amazonaws.com",
+  "tagging.'${REGION}'.amazonaws.com",
+
+  "mirror.openshift.com",
+  "storage.googleapis.com",
+  "api.openshift.com",
+
+  "api.pagerduty.com",
+  "events.pagerduty.com",
+  "api.deadmanssnitch.com",
+  "nosnch.in",
+  "osdsecuritylogs.splunkcloud.com",
+  "http-inputs-osdsecuritylogs.splunkcloud.com",
+  "sftp.access.redhat.com"
+')
+
+DOMAIN_LIST=${OCP_DOMAIN_ALLOW_LIST}
+if [[ ${MANAGED_CLUSTER} == "yes" ]]; then
+  DOMAIN_LIST=${OSD_DOMAIN_ALLOW_LIST}
+fi
+
+function aws_add_param_to_json() {
+    local k="$1"
+    local v="$2"
+    local param_json="$3"
+    if [ ! -e "$param_json" ]; then
+        echo -n '[]' > "$param_json"
+    fi
+    cat <<< "$(jq  --arg k "$k" --arg v "$v" '. += [{"ParameterKey":$k, "ParameterValue":$v}]' "$param_json")" > "$param_json"
+}
+
+# Create DNS firewall
+UNIQUE_HASH=$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c 4)
+STACK_NAME="${NAMESPACE}-${UNIQUE_HASH}-dns-firewall"
+dns_firewall_domain_list_name="${STACK_NAME}-domain-list"
+dns_firewall_rule_group_name="${STACK_NAME}-rule-group"
+RULE_GROUP_ALLOWED=$(echo -e '
+{
+    "AWSTemplateFormatVersion": "2010-09-09",
+    "Description": "",
+    "Resources": {
+        "ResolverFirewallDomainList": {
+            "Type": "AWS::Route53Resolver::FirewallDomainList",
+            "Properties": {
+                "Name": "'${dns_firewall_domain_list_name}'",
+                "Domains": ['${DOMAIN_LIST}']
+            }
+        },
+        "ResolverFirewallRuleGroup": {
+            "Type": "AWS::Route53Resolver::FirewallRuleGroup",
+            "Properties": {
+                "Name": "'${dns_firewall_rule_group_name}'",
+                "FirewallRules": [
+                    {
+                        "Priority": 1,
+                        "Action": "ALLOW",
```

Comment:
I believe we need:
Priority 101: ALLOW rule to allow DNS domains in the white list.
Priority 102: BLOCK_ALL rule to deny all other URLs.

Reply:
Will update Priority to 101 for ALLOW rule. If in the future, we need the BLOCK rule, we need to add a BLOCK ResolverFirewallDomainList.

Reply:
> we need to add a BLOCK ResolverFirewallDomainList.

Do you mean _accept all but deny domains in the black list_?

Reply:
No, what I said is as same as what you did in your CF, adding another domain list `FirewallAllDomainList`  for BLOCK.